### DISCUSSION TITLE
What's the best way to handle one-off components?

### DISCUSSION THREAD

**Discussion by davidturnbull** [markdoc/markdoc#219] [Q&A] [answered]
I'm interested in using Markdoc for documentation and was wondering if there were any recommend patterns for using one-off components? I want to add components to certain pages without bloating the entire site's bundle size.

**Comment by nkohari-stripe** [comment, marked as answer]
The best way to handle this would be via code splitting and asynchronous imports. If you're using React, the most straightforward way to handle this is with [React.lazy](https://reactjs.org/docs/code-splitting.html#reactlazy).

You'll also need to configure your bundler to split the asynchronously-loaded modules into separate JS files during build; the exact approach there depends on which bundler you're using.
